pkg/zcache/local_cache.go

```diff
@@ -178,34 +187,48 @@ func (c *localCache) cleanupExpiredKeys() {
 			continue
 		}
 
+		totalResident++
+
 		if cachedItem.IsExpired() {
 			keysToDelete = append(keysToDelete, entry.Key())
 		}
 
 		if len(keysToDelete) >= c.batchSize {
-			c.deleteKeysInBatch(keysToDelete)
+			totalDeleted += c.deleteKeysInBatch(keysToDelete)
 			keysToDelete = keysToDelete[:0]
 			time.Sleep(c.throttleTime)
 		}
 	}
 
 	if len(keysToDelete) > 0 {
-		c.deleteKeysInBatch(keysToDelete)
+		totalDeleted += c.deleteKeysInBatch(keysToDelete)
 	}
+
+	// update metrics
+	_ = c.metricsServer.UpdateMetric(cleanupItemCountMetricKey, float64(totalResident-totalDeleted), residentItemCountLabel)
+	_ = c.metricsServer.UpdateMetric(cleanupItemCountMetricKey, float64(totalDeleted), deletedItemCountLabel)
```

Comment:
This should be done on two different metrics. Let's create one for deletion, and another one for resident items. By doing so, we will be able to display both values in different charts.